### PR TITLE
MOB-3403 [Branding Capabilities] Improve customization on Android UI …

### PR DESCRIPTION
**Pull request by mateid-kr**
…sdk and implement the described palette to Farechoice App

## JIRA
[MOB-3403](https://karhoo.atlassian.net/browse/MOB-3403)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [X] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [X] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

**Review by cosminBadulescu — DISMISSED** [review]

File: uisdk/src/main/res/drawable/uisdk_gradient_approved.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
     android:shape="rectangle">
     <gradient
         android:angle="0"
-        android:endColor="@color/lightGrey"
-        android:startColor="@color/lightGrey"
+        android:endColor="@color/khTransparentSecondaryColor"
```
Comment (mohadian) [inline uisdk/src/main/res/drawable/uisdk_gradient_approved.xml:5]: I suggest using `KhButtonEnabledColor`

Change the name of the file to `uisdk_gradient_disabled.xml `

File: uisdk/src/main/res/drawable/uisdk_gradient_enabled.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
     android:shape="rectangle">
     <gradient
         android:angle="0"
-        android:endColor="@color/primary"
-        android:startColor="@color/primary"
+        android:endColor="@color/secondary"
```
Comment (mohadian) [inline uisdk/src/main/res/drawable/uisdk_gradient_enabled.xml:5]: I suggest using a similar approach as the previous one`KhButtonDisabledColor`

**Review by mohadian — CHANGES_REQUESTED** [review]
There is an issue with the title ... can you see which change cause this issue? 
Screenshot 2021-06-30 at 21 32 22

**Comment by mohadian** [comment]
It seems we use the same color for the status bar and app bar, in material designs they are usually different colors: 

Screenshot 2021-06-30 at 21 37 27


Screenshot 2021-06-30 at 21 34 53

**Comment by mateid-kr** [comment]
> It seems we use the same color for the status bar and app bar, in material designs they are usually different colors:
> 
> Screenshot 2021-06-30 at 21 37 27
> 
> Screenshot 2021-06-30 at 21 34 53

This is according to the specs:

Use of Primary color
The primary color is used for display components without clickable interaction
Tool/network bar
Display important information (tag, etc)

File: uisdk/src/main/res/drawable/uisdk_gradient_enabled.xml (hunk 2 above)
Comment (mateid-kr) [reply uisdk/src/main/res/drawable/uisdk_gradient_enabled.xml:5]: Done

File: uisdk/src/main/res/drawable/uisdk_gradient_approved.xml (hunk 1 above)
Comment (mateid-kr) [reply uisdk/src/main/res/drawable/uisdk_gradient_approved.xml:5]: Done

File: uisdk/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -318,4 +326,19 @@
         <item name="actionText">@style/Text.Action.Primary</item>
     </style>
 
+    <style name="KhAddressMapPickupIcon">
+        <item name="srcCompat">@drawable/uisdk_ic_dropoff_pin</item>
+    </style>
+
+    <style name="KhBookButton">
```
[the remark this answers is not in the document]
Comment (mateid-kr) [reply uisdk/src/main/res/values/styles.xml:333]: Done. About the Rides being pink, gonna correct it now. It seems it inherited the same style as the one from the toolbar in the map activity

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/address/TimePickerTitleView.kt [hunk 4]
```diff
@@ -16,7 +16,7 @@ class TimePickerTitleView @JvmOverloads constructor(context: Context,
     fun setTitle(@StringRes title: Int, timezone: String): View {
         val pickerTitle = "${resources.getString(title)} ($timezone)"
         text = pickerTitle
-        setBackgroundColor(ContextCompat.getColor(context, R.color.secondary))
+        setBackgroundColor(ContextCompat.getColor(context, R.color.khTimePickerColor))
```
[the remark this answers is not in the document]
Comment (mateid-kr) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/address/TimePickerTitleView.kt:19]: Done

File: uisdk/src/main/res/values/styles.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mateid-kr) [reply uisdk/src/main/res/values/styles.xml:333]: Also, the network bar doesn't have the primary color, will correct this too

**Review by mohadian — APPROVED** [review]

**Pull request merged**